modules/nw-rfhe-creating-hardware-event.adoc

```diff
@@ -0,0 +1,77 @@
+// Module included in the following assemblies:
+//
+// * networking/using-rfhe.adoc
+
+:_content-type: PROCEDURE
+[id="nw-rfhe-creating-hardware-event_{context}"]
+= Creating the HardwareEvent custom resource
+
+To start using baremetal hardware events, create the a `HardwareEvent` CR for the host where the Redfish hardware is present. Hardware events and faults are reported in the `hw-event-proxy` logs.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+* Install the Baremetal Hardware Event Proxy Operator.
+
+[NOTE]
+====
+Create a single instance of the `HardwareEvent` resource. Multiple `HardwareEvent` resources are not permitted.
+====
+
+.Procedure
+
+. Create the `HardwareEvent` custom resource (CR):
+
+.. Save the following YAML in the `hw-event.yaml` file:
++
+[source,yaml]
+----
+apiVersion: "event.redhat-cne.org/v1alpha1"
+kind: "HardwareEvent"
+metadata:
+  name: "hardware-event"
+spec:
+  nodeSelector:
+    node-role.kubernetes.io/hw-event: "" <1>
+  transportHost: "amqp://amq-router-service-name.amq-namespace.svc.cluster.local" <2>
+  logLevel: "debug" <3>
+  msgParserTimeout: "10" <4>
+----
+<1> Mandatory. Use the `nodeSelector` field to target nodes with the specified label, for example, `node-role.kubernetes.io/hw-event: ""`.
+<2> Mandatory. AMQP host that delivers the events at the transport layer using the AMQP protocol.
+<3> Optional. The default value is `debug`. Sets the log level in hw-event-proxy logs. The following log levels are available: `fatal`, `error`, `warning`, `info`, `debug`, `trace`.
+<4> Optional. Sets the timeout value in milliseconds for the Message Parser. If a message parsing request is not responded to within the timeout duration, the original hardware event message is passed to the cloud native event framework. The default value is 10.
+
+.. Create the `HardwareEvent` CR:
++
+[source,terminal]
+----
+$ oc create -f hardware-event.yaml
+----
+
+. Create a BMC username and password `Secret` CR that allows the hardware events proxy to access the Redfish message registry for the bare-metal host.
++
+.. Save the following YAML in the `hw-event-bmc-secret.yaml` file:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: redfish-basic-auth
+type: Opaque
+data:
+  username: <bmc_username>
+  password: <bmc_password>
+stringData:
+  # BMC host DNS or IP address
+  hostaddr: <bmc_host_ip_address>
+----
```

Comment:
The username and password in data field have to be base64 encoded. This might confuse inexperienced user. I think it is easier to just put plain text under stringData instead. Like this:

```
apiVersion: v1
kind: Secret
metadata:
  name: redfish-basic-auth
type: Opaque
stringData:
  username: <bmc_username>
  password: <bmc_password>
  # BMC host DNS or IP address
  hostaddr: <bmc_host_ip_address>
```